crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -50,9 +83,18 @@ impl<T: WasiConfig> Default for WasmtimeEngine<T> {
 
 /// Data that contains both wasi_preview1 and wasi_preview2 contexts.
 pub struct WasiCtx {
-    pub(crate) wasi_preview2: wasi_preview2::WasiCtx,
-    pub(crate) wasi_preview1: wasi_preview1::WasiP1Ctx,
+    // TODO: split WasiCtx into structs that holds only target relevant data
+    pub(crate) wasi_preview1: Option<wasi_preview1::WasiP1Ctx>,
+    pub(crate) wasi_preview2_cli: wasi_preview2::WasiCtx,
+    pub(crate) wasi_preview2_http: wasmtime_wasi_http::WasiHttpCtx,
```

Comment:
should all these fileds be `Option<...>`?

Reply:
If those are all optional it would be awkward to implement `WasiView` and `WasiHttp` traits on the type. But it is now split into two different structs.